### PR TITLE
Path Traversal in Superstatic

### PR DESCRIPTION
**Pull request by buggycoder-sys**
## ✍️ Description

I would like to report path traversal vulnerability in Superstatic module
It allows an attacker to read system files via path traversal vulnerability

Module
-------------------------------------------
module name: Superstatic
version: 6.0.4
npm page: https://www.npmjs.com/package/superstatic

Module Description

Superstatic is an enhanced static web server that was built to power. It has fantastic support for HTML5 pushState applications, clean URLs, caching, and many other goodies.
By default, Superstatic will simply serve the current directory

Weekly Downloads
-------------------------------------------
335,788

Vulnerability Description
-------------------------------------------
With a symbolically linked file in the working directory, it is possible to read arbitrary files outside of the web root directory.

## 🕵️‍♂️ Proof of Concept
Steps To Reproduce:
-------------------------------------------

1)Install the Superstatic module
$ npm install -g superstatic

2)Make a directory
$ mkdir test

3)Go to 'test' directory
$ cd test

4)create a symlink file
ln -s /etc/passwd 'filename'

5)Run Superstatic module
$ Superstatic

6)Request the file within browser
http://localhost:3474/'filename'

Supporting Material/References:
-------------------------------------------

These are similar reports
https://hackerone.com/reports/510043
https://hackerone.com/reports/695416

## 💥 Impact

This could have enabled an attacker to view system files and leverage attacks like remote code execution and so on

## ☎️ Contact
No

## ✅ Checklist

**In my pull request, I have:**

- [x] _Created and populated the `README.md` and `vulnerability.json` files_
- [x] _Provided the repository URL and any applicable [permalinks]([https://help.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files](https://help.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files))_
- [x] _Defined all the applicable weaknesses ([CWEs]([https://cwe.mitre.org/](https://cwe.mitre.org/)))_
- [x] _Proposed the CVSS vector items i.e. User Interaction, Attack Complexity_
- [x] _Checked that the vulnerability affects the latest version of the package released_
- [x] _Checked that a fix does not currently exist that remediates this vulnerability_
- [x] _Complied with all applicable laws_

### CODE REVIEW

**Comment by buggycoder-sys** [comment]
Thank you for the verification @mufeedvh and @Mik317 :)

**Comment by Mik317** [comment]
@buggycoder-sys, it's been a pleasure :+1: ,
looking forward to seeing new disclosure and fixes by you :smile: 

Cheers,
Mik

**Comment by buggycoder-sys** [comment]
@Mik317 Sure <3

**Comment by buggycoder-sys** [comment]
@Mik317 Sure ❤️

**Comment by buggycoder-sys** [comment]
Done @JamieSlome  :)

File: bounties/npm/Superstatic/1/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+## Description
```
Comment (JamieSlome) [inline bounties/npm/Superstatic/1/README.md:1]: Single `#` like `# Description`.

File: bounties/npm/Superstatic/1/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+Weekly Downloads
+-------------------------------------------
+335,788
+
+Vulnerability Description
+-------------------------------------------
+With a symbolically linked file in the working directory, it is possible to read arbitrary files outside of the web root directory.
+
+
+## POC
```
Comment (JamieSlome) [inline bounties/npm/Superstatic/1/README.md:26]: Single `#` like `# PoC`.

File: bounties/npm/Superstatic/1/vulnerability.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
+{
+    "PackageVulnerabilityID": "1",
+    "DisclosureDate": "2020-08-07",
+    "AffectedVersionRange": "*",
+    "Summary": "Path Traversal",
+    "Discloser": {
```
Comment (JamieSlome) [inline bounties/npm/Superstatic/1/vulnerability.json:6]: This object should be:

```
"Contributor": {
   "Discloser": "buggycoder-sys",
   "Fixer": ""
}
```

You only need to provide your GitHub username and leave the fixer empty.

File: bounties/npm/Superstatic/1/vulnerability.json [hunk 4]
```diff
@@ -0,0 +1,52 @@
+{
+    "PackageVulnerabilityID": "1",
+    "DisclosureDate": "2020-08-07",
+    "AffectedVersionRange": "*",
+    "Summary": "Path Traversal",
+    "Contributor": {
+         "Discloser": "buggycoder-sys",
+         "Fixer": ""
+}
```
Comment (JamieSlome) [inline bounties/npm/Superstatic/1/vulnerability.json:9]: Be careful of a missing comma here! 🍰

**Comment by JamieSlome** [comment]
@buggycoder-sys - looks good, just that missing comma left after the `Contributor` JSON object in the `vulnerability.json` and then all LGTM! 👍

**Pull request merged**